docs/concepts/nuclio-real-time-functions.ipynb

```diff
@@ -100,166 +100,122 @@
+    "- {ref}`~mlrun.projects.MlrunProject.delete_api_gateway`\n",
+    "- {ref}`~mlrun.projects.MlrunProject.get_api_gateway`\n",
+    "- {ref}`~mlrun.projects.MlrunProject.list_api_gateways`\n",
     "\n",
     "\n",
-    "# Enter your username and password here\n",
-    "username = \"username\"\n",
-    "password = \"password\"\n",
+    "### Create an API gateway with basic authorization\n",
     "\n",
-    "# Enter your API Gateway endpoint here\n",
-    "basic_auth_api_gateway_path = \"https://<API GATEWAY ENDPOINT>\"\n",
+    "Assume you already deployed a nuclio function:\n",
     "\n",
-    "headers = {\"Authorization\": basic_auth(username, password)}\n",
-    "res = requests.get(url=basic_auth_api_gateway_path, headers=headers, verify=False)\n",
-    "print(res.text)"
+    "```\n",
+    "fn = mlrun.code_to_function(\n",
+    "            filename=\"nuclio_function.py\",\n",
+    "            name=f\"nuclio-func\",\n",
+    "            kind=\"nuclio\",\n",
+    "            image=\"python:3.9\",\n",
+    "            handler=\"handler\",\n",
+    "        )\n",
+    "```\t"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "f5941683",
+   "id": "2befaac3",
    "metadata": {},
    "source": [
-    "### Making an HTTP request using access key authentication\n",
-    "1. Create an API Gateway in the UI, with authentication `access key` and choose the `access-key` Nuclio function.\n",
-    "2. Give it a name and copy the endpoint.\n",
-    "2. Paste the endpoint after the `https://`.\n",
-    "2. In the UI, click the user's top right icon, then copy the access key from there.\n",
-    "2. Change the access key in the code below."
+    "Define the API gateway entity:\n",
+    "\n",
+    "```\n",
+    "my_api_gateway = mlrun.runtimes.nuclio.api_gateway.APIGateway(\n",
+    "            mlrun.runtimes.nuclio.api_gateway.APIGatewayMetadata(\n",
+    "                name=\"gw-with-basic-auth\",\n",
+    "            ),\n",
+    "            mlrun.runtimes.nuclio.api_gateway.APIGatewaySpec(\n",
+    "                functions=fn,\n",
+    "                project=project.name,\n",
+    "            ),\n",
+    "    )\n",
+    "```\n",
+    "Add basic authorization configuration:\n",
+    "```\n",
+    "my_api_gateway.with_basic_auth(username=\"test\",\n",
+    " password=\"pass\")\n",
+    "```\n",
+    "Create (or update) the API gateway</br>\n",
+    "It's crucial to update the defined API Gateway entity with the entity returned from store_api_gateway method. This fills in important fields like host/path, etc.\n",
+    "```\n",
+    "my_api_gateway = project.store_api_gateway(my_api_gateway)\n",
+    "```\n",
+    "store_api_gateway checks that api_gateway is ready to be invoked before returning it. You can also check if api gateway is in ready state:\n",
+    "```\n",
+    "my_api_gateway.is_ready()\n",
+    "```\n",
+    "\n",
+    "If you know that the API gateway was changed (for example, in the UI), load the latest changes:\n",
```

Comment:
After implementing this task https://iguazio.atlassian.net/browse/ML-6491, we don't allow changing GW from UI. So, it is better to say smth like `(for example, in another jupyter notebook)`

Also, we need to add warning that any change of api gateway from Nuclio UI will break everything, so we don't allow this.

Reply:
@rokatyy we don't really enforce it, so `don't allow` is a bit strong.
It is **discouraged** because the the functionality will break.